### DISCUSSION TITLE
Getting only a subset of field values from listitems

### DISCUSSION THREAD

**Discussion by luciusbv** [pnp/pnpcore#551] [Q&A] [answered]
Hi,

In the code:
_var configurationList = await context.Web.Lists.GetByTitleAsync(listName, p => p.Title);
var mappingListItems = configurationList.Items;_

I'm getting all the items from a small list. However,  I'm interested only in the values of 2 distinct fields. So, I tried to find out I can limit the set of returned list items fields, for minimizing the footprint of the request. I was expecting something  like the code below to work: 
_var configurationList = await context.Web.Lists.GetByTitleAsync(listName, p => p.Title);
var mappingListItems = configurationList.Items.QueryProperties(i => i.Id, i => i.Title, i => i.Properties[mappingFieldName], i => i.Properties[choiceFieldName]);_

That failed to provide the expected result. Also, I tried to find an answer in the existent documentation and videos, without success.
I already know I can achieve this by using instead, configurationList.LoadItemsByCamlQueryAsync but I would really like to stay out of caml queries for simple scenarios as described above.

Any help on this will be really appreciated,
  Thank you!

**Comment by jansenbe** [comment, marked as answer]
Hi @luciusbv ,

We currently don't support scoping the request for doing a regular list item load, but that's definitely an idea to consider for adding. In the meanwhile I suggest you use the ListDataAsStream approach (https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#using-the-listdataasstream-approach), create a simple ViewXml requesting the fields to load.

**Reply by wonderplayer** [reply]
Hi @jansenbe 
Wondering if something like querying list item properties on load was implemented 😊

**Reply by jansenbe** [reply]
@wonderplayer : if you want to control the returned fields it's best to use either `LoadItemsByCamlQuery` or `ListDataAsStream` as explained here: https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#you-need-to-read-system-properties-like-fileleafref-filedirref-or-you-need-to-filter-list-items-or-you-want-to-define-the-returned-fields

**Reply by wonderplayer** [reply]
I see. Hoped that I missed something 😄
Thanks!
